articles/getting-started-with-jwt-using-angular8-and-nodejs/index.md

```diff
@@ -0,0 +1,293 @@
+    var userData = {
+        "name": "My Name",
+        "id": "1234"
+    }
+    let token = jwt.sign(userData, secret, { expiresIn: '15s'})
+    res.status(200).json({"token": token});
+});
+```
+Once the route has been created, it stores the user data and encodes the data. When decoded, it will give back the user data, in this case only username and id, since storing a password is never a good practice. Next will be to run the application and check the token generated by running:
+```bash
+$ node app.js
+```
+In this guide, I will use Postman to test the routes as shown below.
+
+![first web token](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/token-generation1.png)
+
+As of now, I have generated the first web token. To illustrate this, I will create a `/path1` and secure it using my JSON Web Token.
+
+For this task, I will be using the express-jwt function.
+```javascript
+app.get('/path1', (req, res) => {
+    res.status(200)
+        .json({
+            "success": true,
+            "msg": "Secrect Access Granted"
+        });
+});
+```
+Using expressJWT function:
+```javascript
+//This is to allow access to the path with no token authentication
+app.use(expressJWT({ secret: secret, algorithms: ['HS256']})
+    .unless(
+        { path: [
+            '/token/sign'
+        ]}
+    ));
+```
+Next, I will be testing the path using Postman, with no token sent in the header.
+
+![401 Error](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/401-unauthorized.png)
+
+The app did not allow me to access the path, the error `401 Unauthorized` was displayed as expected. The code below will use the token I first obtained from the `/token/sign-in` route.
+```javascript
+app.get('/path1', (req, res) => {
+    res.status(200)
+        .json({
+            "success": true,
+            "msg": "Secret Access Granted"
+        });
+});
+```
+On adding the Bearer Token, the success message is displayed as shown below:
+
+![success message](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/success-token.png)
+
+Back to Angular, I will create a new component `home` by running the following command:
+```bash
+$ ng generate component home
+```
+Then modify `home.component.ts` located at `src/app` to look as shown below:
+```javascript
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.css']
+})
+export class HomeComponent implements OnInit {
+
+  readonly API_URL = 'http://localhost:3000';
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  ngOnInit() {
+  }
+
+signIn() {
+    this.http.get(this.API_URL + '/token/sign')
+      .subscribe(
+        (res) => {
+          console.log(res);
+          if (res['token']) {
+            localStorage.setItem('token', res['token']);
+          }
```

Comment:
Please explain what these pieces of code do in detail. Explain your code line by line if possible to bring more understanding to the reader. Please go through all your code blocks explaining your code.